spec/requests/container_images_spec.rb

```diff
@@ -45,4 +45,41 @@
       expect(response.parsed_body).to include(expected)
     end
   end
+
+  context 'POST /api/container_images/scan' do
+    let(:provider) { FactoryGirl.create(:ems_kubernetes) }
+    let(:container_image) { FactoryGirl.create(:container_image, :ext_management_system => provider) }
+    let(:conatiner_id) { container_image.id }
+    let(:invalid_container_image) { container_image.id + 100 }
+    let(:invalid_image_url) { api_provider_container_image_url(nil, provider, invalid_container_image) }
+
+    it "responds with 404 Not Found for an invalid container image" do
+      api_basic_authorize(action_identifier(:container_images, :scan, :subresource_actions, :post))
+
+      post(invalid_image_url, :params => { :action => "scan" })
+
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it "doesn't scan a Container Image without appropriate role" do
+      api_basic_authorize
+
+      post(invalid_image_url, :params => { :action => "scan" })
```

Comment:
You might want a valid URL to ensure you're only changing one thing here, relative to the happy path?